yarl/_url.py

```diff
@@ -233,6 +233,8 @@ def build(
             raise ValueError(
                 'Can\'t mix "authority" with "user", "password", "host" or "port".'
             )
+        if isinstance(port, str) and port == "":
```

Comment:
How about
```suggestion
        if isinstance(port, str) and not port:
```

Reply:
Or even better:
```suggestion
        if port == "":
```
(you don't really need to check the type, since comparing to a string literal would be falsy if it's not)